### PR TITLE
Test authentication.

### PR DESCRIPTION
**Pull request by ltratt**
In a sense these tests are the two most important missing tests, because they're what give confidence that snare is only executing commands when the request is properly authenticated.

### CODE REVIEW

File: tests/basic.rs [hunk 1]
```diff
@@ -177,3 +177,128 @@ github {{
...
+        exit_success,
+    )
+}
+
+#[test]
+fn wrong_secret() -> Result<(), Box<dyn Error>> {
+    // Takes the example from [full_request], alters the client-side secret, and checks that this
+    // causes snare not execute any commands (so, by proxy, we assume that authentication failed).
+
+    let td = Builder::new().tempdir_in(env!("CARGO_TARGET_TMPDIR"))?;
```
Comment (vext01) [inline tests/basic.rs:248]: Let's lift the common setup into a function?

File: tests/basic.rs [hunk 2]
```diff
@@ -177,3 +177,128 @@ github {{
...
+        &format!(
+            r#"listen = "127.0.0.1:28085";
+github {{
+  match ".*" {{
+    cmd = "touch {tps}";
+    secret = "secretsecret";
+  }}
+}}"#
+        ),
+        move |_| {
```
Comment (vext01) [inline tests/basic.rs:222]: I'm struggling to find the bit where you check that it failed.

File: tests/basic.rs (hunk 1 above)
Comment (ltratt) [reply tests/basic.rs:248]: The next PR will do just that :)

File: tests/basic.rs (hunk 2 above)
Comment (ltratt) [reply tests/basic.rs:222]: We (indirectly, which is the best we can do) check that the action wasn't executed (i.e. the named temporary file wasn't created) in https://github.com/softdevteam/snare/pull/123/files#diff-e41c063cd26a1181211ca22e91577e32d5067e01cf8f6341954e708842bf5f07R296.

Comment (vext01) [reply tests/basic.rs:222]: I see.

**Pull request merged**